airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/source/jdbc/AbstractJdbcSource.kt

```diff
@@ -800,4 +813,27 @@ abstract class AbstractJdbcSource<Datatype>(
             return result
         }
     }
+
+    override fun createReadIterator(
+        database: JdbcDatabase,
+        airbyteStream: ConfiguredAirbyteStream,
+        catalog: ConfiguredAirbyteCatalog?,
+        table: TableInfo<CommonField<Datatype>>,
+        stateManager: StateManager?,
+        emittedAt: Instant
+    ): AutoCloseableIterator<AirbyteMessage> {
+        val iterator =
+            super.createReadIterator(
+                database,
+                airbyteStream,
+                catalog,
+                table,
+                stateManager,
+                emittedAt
+            )
+        return when (airbyteStream.syncMode) {
+            INCREMENTAL -> augmentWithStreamStatus(airbyteStream, iterator)
```

Comment:
curious: Why not move `augmentWithStreamStatus` for full refresh here too?

Reply:
This was an edge case I discovered in which an incremental sync falls back to  full refresh syncing in case the cursor column is not found in catalog.
In that case we do not augment the iterator with statuses as full refresh streams augment it separately in a different place.
A failing  test led me to discover this edge case